test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,160 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+
+	ctx.TargetPodsCount = nonHDFSTargetPodCount + 1
+
+	namespace, errTwo := createTestingNamespace(ctx)
+	if errTwo != nil {
+		return errTwo
+	}
+
+	//decode and create yamls except for persistentvolume
+
+	type nfsManifestItem struct {
+		resourceType interface{}
+		resourceFile string
+		actions      []string
+	}
+	type nfsServer struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsService struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsStorageclass struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+	var nfsManifestItems = map[string]nfsManifestItem{
+		"nfsServer":
+		{
+
+			resourceType: nfsServer{},
+			resourceFile: "/test/testhelpers/nfs_server.yaml",
+			actions:      []string{"CREATE"},
+		},
+		"nfsService":
+		{
+			resourceType: nfsService{},
+			resourceFile: "/test/testhelpers/nfs_service.yaml",
+			actions:      []string{"CREATE", "GET"},
+		},
+		"nfsStorageclass":
+		{
+			resourceType: nfsStorageclass{},
+			resourceFile: "/test/testhelpers/nfs_storageclass.yaml",
+			actions:      []string{"CREATE"},
+		},
+	}
+
+	for _, manifest := range nfsManifestItems {
+
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsServer"].resourceFile, nfsManifestItems["nfsServer"].resourceType)
+		err = deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsService"].resourceType)
+		err = deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsStorageclass"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceType)
+		if err != nil {
+			return fmt.Errorf("failed to decode the %s YAML service manifest: %v", err, manifest.resourceFile)
+		}
+
+		for _, action := range manifest.actions {
+			if action == "CREATE" {
+				ctx.Client.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"], nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+				_, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
```

Comment:
The whole idea behind using a loop is to handle the process of decoding and creating these manifests in a single iteration. Currently, you're making the call to decode several manifests inside a single loop iteration.